### PR TITLE
#93 mechanical valve driver

### PR DESCRIPTION
**Pull request by thejoelm**

### CODE REVIEW

**Review by IonLi09 — DISMISSED** [review]
lgtm

**Comment by panyab** [comment]
Software hasn't implemented a command for manual valve opening, so that shouldn't be there. Also, you need to register VALVE_COMMAND in the EosLib/lib/mqtt/init.py file.

**Review by DarylDohner — CHANGES_REQUESTED** [review]
incomplete, hasnt been tested, see panya's comment &
https://vip-les.slack.com/archives/C05RS9BJKPA/p1699388426129139
btw you need a config.json entry for this driver

**Review by aryanbattula — COMMENTED** [review]
lgtm

File: valve_driver.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        self._logger.info("~~~OPENING VALVE~~~")
+        GPIO.output(ValveDriver.valve_pin, GPIO.HIGH)
+        self.thread_sleep(self._logger, ValveDriver.time_valve_open)
+        GPIO.output(ValveDriver.valve_pin, GPIO.LOW)
+        self._logger.info("~~~CLOSING VALVE~~~")
+
+    @staticmethod
+    def valve_trigger_mqtt(client, user_data, message):
+        user_data['logger'].info("Received valve open command")
+        user_data['queue'].put(1)
```
Comment (DarylDohner) [inline valve_driver.py:85]: this seems unintentional, probably want to remove this since the actual enqueing is done below

**Review by DarylDohner — CHANGES_REQUESTED** [review]

**Review by aryanbattula — APPROVED** [review]
lgtm!

**Pull request merged**